src/Core/Services/MetadataProviders/MetadataProviderFactory.cs

```diff
@@ -62,7 +62,10 @@ private void ConfigureMetadataProviders()
 
         public void OnConfigChanged(object? sender, HotReloadEventArgs args)
         {
+            _metadataProviders = new Dictionary<string, ISqlMetadataProvider>();
```

Comment:
better to clear() because it avoids re-allocating the _metadataProviders dictionary every time the OnConfigChanged method is called. it is more efficient in terms of memory and performance.

I'll address it in my PR. so you can remove revert this